plugins/woocommerce/src/Admin/Features/ProductBlockEditor/Init.php

```diff
@@ -56,6 +57,13 @@ public function __construct() {
 		$this->redirection_controller = new RedirectionController( $this->supported_post_types );
 
 		if ( \Automattic\WooCommerce\Utilities\FeaturesUtil::feature_is_enabled( 'product_block_editor' ) ) {
+			add_filter( 'product_type_selector', function ( $types ) {
+				$types['simple-subscription'] = __( 'Simple subscription', 'woocommerce' );
+				$types['variable-subscription'] = __( 'Variable subscription', 'woocommerce' );
+				return $types;
+			}, 999, 1 );
+			add_filter( 'woocommerce_product_editor_product_templates', array( $this, 'get_product_templates' ), -999, 1 );
```

Comment:
Instead of doing this, you should be passing the result of `$this->get_product_templates()` into the `apply_filters` call below. That will set the default value of `$templates` that gets passed to any hooks.

Reply:
Ohh! Nice, you are right!